pkg/prometheus/operator_test.go

```diff
@@ -394,3 +394,73 @@ func TestValidateRelabelConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateProberUrl(t *testing.T) {
+	for _, tc := range []struct {
+		proberSpec     monitoringv1.ProberSpec
+		validProberURL bool
+	}{
+		{
+			proberSpec: monitoringv1.ProberSpec{
+				URL: "http://blackbox-exporter.example.com",
+			},
+			validProberURL: false,
+		},
+		{
+			proberSpec: monitoringv1.ProberSpec{
+				URL: "https://blackbox-exporter.example.com",
+			},
+			validProberURL: false,
+		},
+		{
+			proberSpec: monitoringv1.ProberSpec{
+				URL: "ftp://fileserver.com",
+			},
+			validProberURL: false,
+		},
+		{
+			proberSpec: monitoringv1.ProberSpec{
+				URL: "192.168.178.3",
+			},
+			validProberURL: true,
+		},
+		{
+			proberSpec: monitoringv1.ProberSpec{
+				URL: "192.168.178.3:9090",
```

Comment:
Hm, how does this test case pass if the validation only checks whether the first part is a host?

Reply:
It checks either hostname or Ip https://github.com/asaskevich/govalidator/blob/f21760c49a8d602d863493de796926d2a5c1138d/validator.go#L851-L853